platform/ios/src/MGLMapView.mm

```diff
@@ -750,6 +753,20 @@ - (void)glkView:(__unused GLKView *)view drawInRect:(__unused CGRect)rect
         _mbglMap->render();
 
         [self updateUserLocationAnnotationView];
+        
+        // Hide all annotation views
+        for (MGLAnnotationView *annotationView in _annotationViews) {
+            annotationView.hidden = YES;
+        }
+        
+        // Update visible annotation views
+        std::vector<MGLAnnotationTag> annotationTags = [self annotationTagsInRect:CGRectInset(self.bounds, 0, 0)];
+        for (std::size_t i = 0; i != annotationTags.size(); ++i) {
```

Comment:
I'm sure I'm missing something! 😄 

But, I do want to loop over the vector result of `annotationTagsInRect:` and not the the entire contents of `_annotationContextsByAnnotationTag`. In the spirit of not conjuring up the index when I don't need it, how about:

```
std::vector<MGLAnnotationTag> annotationTags = [self annotationTagsInRect:CGRectInset(self.bounds, 0, 0)];
for(auto const& annotationTag: annotationTags)
 {
    MGLAnnotationContext &annotationContext = _annotationContextsByAnnotationTag[annotationTag];
...
```